framework/src/main/java/org/checkerframework/framework/util/ContractsFromMethod.java

```diff
@@ -140,6 +142,27 @@ private <T extends Contract> Set<T> getContractsOfKind(
       }
     }
 
+    // At this point, `result` contains only framework-defined contract annotations.
+    Set<AnnotationMirror> tops = factory.getQualifierHierarchy().getTopAnnotations();
+    for (T contract : result) {
+      AnnotationMirror anno = contract.annotation;
+      if (AnnotationUtils.containsSame(tops, anno)) {
+        // TODO: issue a warning on the annotation itself rather than on the method declaration.
+        // This requires iterating over the annotation trees on the method declaration to determine
+        // which one led tho the given AnnotationMirror.
+        if (executableElement != null) {
+          String methodString = " on method " + executableElement.getSimpleName();
+          factory
+              .getChecker()
+              .reportWarning(
```

Comment:
Warnings should only be issued in the BaseTypeVisitor.  Specifically, this code belongs in `BaseTypeVisitor#checkContractsAtMethodDeclaration`.